service/src/PlatformIO.cpp

```diff
@@ -438,17 +504,38 @@ namespace geopm
         }
 
         double result = NAN;
-        auto iogroup = find_signal_iogroup(signal_name);
-        if (iogroup == nullptr) {
+        auto iogroups = find_signal_iogroup(signal_name);
+        if (iogroups.empty()) {
             throw Exception("PlatformIOImp::read_signal(): signal name \"" + signal_name + "\" not found",
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
-        else if (iogroup->signal_domain_type(signal_name) != domain_type) {
-            result = read_signal_convert_domain(signal_name, domain_type, domain_idx);
+
+        std::string err_msg;
+        for (auto ii : iogroups) {
+            if (ii->signal_domain_type(signal_name) != domain_type) {
+                result = read_signal_convert_domain(signal_name, domain_type, domain_idx);
+                break;
+            }
+            else {
+                try {
+                    result = ii->read_signal(signal_name, domain_type, domain_idx);
+                    break;
+                }
+                catch (const geopm::Exception &ex) {
+                    err_msg += std::string(ex.what()) + "\n";
+                }
+            }
         }
-        else {
-            result = iogroup->read_signal(signal_name, domain_type, domain_idx);
+        if (std::isnan(result)) {
```

Comment:
We need a different variable to track if the read was successful.  NAN may be returned when no exception was raised.